test/shared/src/main/scala/zio/test/TestRuntime.scala

```diff
@@ -0,0 +1,108 @@
+package zio.test
+
+import zio.Exit
+import zio.ZIO
+import zio.internal.Executor
+import zio.random.Random
+import zio.stream.ZStream
+
+object TestRuntime {
```

Comment:
I think we need a better name here because at this point we're not doing anything with the `Runtime`. What if we called it `RandomExecutor` and then we could have a trait that extended `Executor` and a companion object with methods for using the `RandomExecutor` along with the rewrite function to generate random execution paths?

Reply:
@adamgfraser the object is now called `RandomExecutor` and has only `paths` and `run` as public methods. Is this API ok?